shell/src/test/java/alluxio/cli/fsadmin/command/SummaryCommandTest.java

```diff
@@ -0,0 +1,193 @@
+    mFile = new File("./MockSummaryResult.txt");
+    FileOutputStream fileOutputStream = new FileOutputStream(mFile);
+    mPrintStream = new PrintStream(fileOutputStream);
+  }
+
+  @After
+  public void cleanDependencies() throws IOException {
+    mMetaMasterClient.close();
+    mBlockMasterClient.close();
+    mPrintStream.close();
+    mFile.delete();
+  }
+
+  @Test
+  public void callSummaryWithDependencies() throws IOException {
+    SummaryCommand summaryCommand = new SummaryCommand(mMetaMasterClient,
+        mBlockMasterClient, mPrintStream);
+    summaryCommand.run();
+    mPrintStream.flush();
+    mPrintStream.close();
+
+    BufferedReader reader = new BufferedReader(new FileReader(mFile));
+    checkIfOutputValid(reader);
+  }
+
+  /**
+   * Checks if the output contains expected values.
+   *
+   * @param reader reader to read SummaryCommand output
+   */
+  private void checkIfOutputValid(BufferedReader reader) throws IOException {
+    StringBuilder outputBuilder = new StringBuilder();
+    String line;
+    while ((line = reader.readLine()) != null) {
+      outputBuilder.append(line);
+    }
+    String output = outputBuilder.toString();
+
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Master Address: " + mAddress));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Web Port: " + mWebPort));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Rpc Port: " + mRpcPort));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Started: " + CommonUtils.convertMsToDate(mStartTimeMs)));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Uptime: " + CommonUtils.convertMsToClockTime(mUpTimeMs)));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Version: " + mVersion));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Safe Mode: " + mSafeMode));
+
+    Assert.assertThat(output, CoreMatchers.containsString("Live Workers: " + mLiveWorkerNum));
+    Assert.assertThat(output, CoreMatchers.containsString("Lost Workers: " + mLostWorkerNum));
+    Assert.assertThat(output, CoreMatchers.containsString("Total Capacity: "
+        + FormatUtils.getSizeFromBytes(mCapacityBytes)));
+    Assert.assertThat(output, CoreMatchers.containsString("Total Capacity: "
+        + FormatUtils.getSizeFromBytes(mCapacityBytes)));
+    Assert.assertThat(output, CoreMatchers.containsString("Used Capacity: "
+        + FormatUtils.getSizeFromBytes(mUsedBytes)));
+    Assert.assertThat(output, CoreMatchers.containsString("Free Capacity: "
+        + FormatUtils.getSizeFromBytes(mFreeBytes)));
+    Assert.assertThat(output, CoreMatchers.containsString("Tier: MEM"
+        + "  Size: " + FormatUtils.getSizeFromBytes(mCapacityBytesOnTiers.get("MEM"))));
+    Assert.assertThat(output, CoreMatchers.containsString("Tier: MEM"
+        + "  Size: " + FormatUtils.getSizeFromBytes(mUsedBytesOnTiers.get("MEM"))));
```

Comment:
can we compare the entire raw output instead of individual components? This way we can make sure the overall appearance is what we want it to be. To make debugging easier, we can compare one line at a time so that we can indicate which line is wrong

Reply:
use Assert.assertEquals to do the explicitly check for each line in the output string.thanks
```java
for (int lineNum = 0; lineNum < outputLines.length; lineNum++) {
      switch (lineNum) {
        case 0:
          Assert.assertEquals("Alluxio Cluster Summary: ", outputLines[lineNum]);
          break;
        case 1:
          Assert.assertEquals(indent("Master Address: " + ADDRESS), outputLines[lineNum]);
          break;
...
```